core-bundle/src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -99,6 +101,32 @@ public function load(array $configs, ContainerBuilder $container): void
         ;
     }
 
+    /**
+     * Validates and sets the "contao.image.sizes" parameter.
+     */
+    private function setPredefinedImageSizes(array $config, ContainerBuilder $container): void
+    {
+        if (!isset($config['image']['sizes'])) {
+            return;
+        }
+
+        $imageSizes = [];
+
+        foreach ($config['image']['sizes'] as $name => $value) {
+            if (preg_match('/^\d+$/', (string) $name)) {
+                throw new \RuntimeException(sprintf('Image size name "%s" cannot contain only digits!', $name));
+            }
+
+            if (in_array($name, [ResizeConfigurationInterface::MODE_BOX, ResizeConfigurationInterface::MODE_PROPORTIONAL, ResizeConfigurationInterface::MODE_CROP], true)) {
+                throw new \RuntimeException(sprintf('Image size name "%s" is reserved and not allowed!', $name));
+            }
+
+            $imageSizes['_' . $name] = $value;
+        }
+
+        $container->setParameter('contao.image.sizes', $imageSizes);
```

Comment:
There should not be a need to store this as a parameter, we should rather inject it into the correct service directly. Something like 

```php
$def = $container->getDefinition('contao.image_factory');
$def->addMethodCall('setPredefinedSizes', $imageSizes);
```